### DISCUSSION TITLE
0.1.0

### DISCUSSION THREAD

**Discussion by xnorpx** [xnorpx/blue-onyx#3] [Announcements]
**Blue Onyx** is an object detection service that processes JPG images, identifies objects, and returns bounding boxes for the detected items. It is fully API-compatible with both **Blue Iris** and **Agent DVR**.  

Blue Onyx is a streamlined refactor of [[Blue Candle](https://github.com/xnorpx/blue-candle)](https://github.com/xnorpx/blue-candle), with several key improvements and changes.  

### Key Differences Between Blue Candle and Blue Onyx:  
- **(Still) No Coral Support**  
- **Intel and AMD GPU Support**  
- **Custom Model Support**: You can use your own trained models.  
- **Inference Backend**: Switched from Candle to ONNX.  
- **Model Update**: Replaced YOLOv8 with [RT-DETR](https://github.com/lyuwenyu/RT-DETR)v2 ([RT-DETR](https://github.com/lyuwenyu/RT-DETR)).  
- **DirectML on Windows**: Uses DirectML instead of CUDA for easier builds and broader support.  
- **Benchmark Tool**: Added a binary to run object detection outside of the service.  
- **Windows-Only Support**: No Linux builds are available.  
- **Simpler Architecture**: Improved design for ease of use.  
- **CLI Updates**: Some command-line interface changes.  

### Additional Information  
More FAQs and detailed documentation will be provided soon.  

---

![Blue Onyx Interface](https://github.com/user-attachments/assets/bb5d2a1f-7845-45d2-979b-3aea65b66fb7)

<hr /><em>This discussion was created from the release <a href='https://github.com/xnorpx/blue-onyx/releases/tag/0.1.0'>0.1.0</a>.</em>

**Comment by loopy321** [comment]
The file 
![image](https://github.com/user-attachments/assets/b75e0d8e-a0fe-49e6-8500-28bfbd729484)
Shows in Windows Defender:
Trojan:Script/Sabsik.FL.A!ml

**Reply by xnorpx** [reply]
That's not good, can you get more information about the trojan without extracting anything? (I cannot reproduce this locally)

**Reply by xnorpx** [reply]
https://en.delphipraxis.net/topic/5470-trojanscriptsabsikteaml-detected-false-positive-of-course/

Looks like I need to buy certificate.

I did try to use: https://www.virustotal.com/

I did not flag the .zip file.

@loopy321 can you try upload the zip file to virustotal.com?

**Reply by loopy321** [reply]
Don't know what happened earlier, but now it downloads without complaint.  In addition, below is the virustotal.com report for the file:
Looks clean.
https://www.virustotal.com/gui/url/707a4e5aaf803da2b3f29655b88ed14674386c6636b36794296642b14320c4d5/detection

**Reply by xnorpx** [reply]
Thanks for circling back! Hopefully it works for you now!

**Reply by xnorpx** [reply]
I have submitted it to Microsoft as a false positive but not sure what they do about it.
